src/main/clojure/clara/rules/engine.cljc

```diff
@@ -1766,6 +1767,11 @@
                     ;; Therefore, the reordering of retractions and insertions should have no impact
                     ;; assuming that the evaluation of rule conditions is pure, which is a general expectation
                     ;; of the rules engine.
+                    (l/fire-activation! listener
+                                        activation
```

Comment:
These extra `deref` calls are going to incur runtime cost. That's sort of an unfortunate part of the listener calls in the engine all being functions (as opposed to macros that could try to mitigate the cost when they are "off").

In this case, I think You could just bind to the `deref`ed values here and use them in the places you need them. They are needed outside of the `l/fire-activation!` call anyways.